pgx/src/spi.rs

```diff
@@ -244,6 +244,21 @@ impl Spi {
         F: FnOnce(SpiClient) -> std::result::Result<Option<R>, SpiError>,
     >(
         f: F,
+    ) -> Option<R> {
+        Self::connect_ext(false, f)
+    }
+
+    /// execute SPI commands via the provided `SpiClient` and return a value from SPI which is
+    /// automatically copied into the `CurrentMemoryContext` at the time of this function call
+    ///
+    /// Setting `nonatomic` to `true` allows transaction control calls (SPI_commit, SPI_rollback).
+    /// Otherwise, calling those functions will result in an immediate error.
```

Comment:
There's no reason to rely on global mutable state for this if we are going to expose a safe wrapper: these kinds of predicates should be enforced at the type level.